### PR TITLE
Cleanup WASM ICU PackageReference usage to use generated property

### PR DESCRIPTION
**Pull request by safern**
This simplifies the way we import ICU package assets into our build and also eliminates the hack to use NuGetPackageRoot.

@akoeplinger I couldn't do the same for llvm-init because of the package name, the property contains the hostarch and os encoded in it and I don't know how to reference a dynamic property name. 

@nkolev92 is there a way to tell NuGet the name of the property I want to be generated for a given PackageReference?

### CODE REVIEW

File: src/mono/mono.proj [hunk 1]
```diff
@@ -1,5 +1,4 @@
-<Project DefaultTargets="Build">
-  <Import Project="Directory.Build.props" />
+<Project Sdk="Microsoft.Build.Traversal" DefaultTargets="Build">
```
Comment (ViktorHofer) [inline src/mono/mono.proj:1]: You are making this project a traversal even though you don't build any projects here (P2Ps). Why is that so? Can't you usehttps://github.com/microsoft/MSBuildSdks/tree/master/src/NoTargets?

**Review by ViktorHofer — APPROVED** [review]

File: src/mono/mono.proj (hunk 1 above)
Comment (safern) [reply src/mono/mono.proj:1]: I tried that but I would need to set a TargetFramework in order to be able to restore the project. Do we want to se TargetFramework for this project?

Comment (ViktorHofer) [reply src/mono/mono.proj:1]: Not really, I guess Traversal is fine then.

Comment (safern) [reply src/mono/mono.proj:1]: Sounds good. Thanks

**Comment by nkolev92** [comment]
@safern 

Nope, you can't control the variable name generated.

**Comment by safern** [comment]
> Nope, you can't control the generated property name.

That would be a really nice feature, I'll open an issue, thanks.

**Review by akoeplinger — COMMENTED** [review]
Awesome, thank you!

**Pull request merged**